pkg/reconciler/revision/resources/deploy.go

```diff
@@ -248,6 +248,18 @@ func MakeDeployment(rev *v1.Revision,
 		return nil, fmt.Errorf("failed to create PodSpec: %w", err)
 	}
 
+	replicaCount := ptr.Int32(autoscalerConfig.InitialScale)
```

Comment:
seems clearer here (and on L259) to just calculate a replica count and then do `Replicas: ptr.Int32(replicaCount)` once in the struct at the bottom